pkg/boot/zbi/zbi.go

```diff
@@ -191,11 +241,13 @@ func (i *Image) readContainerHeader(f io.ReadSeeker) error {
 	return nil
 }
 
+// ZBIKernel abstracts an in-memory kernel entry.
 type ZBIKernel struct {
 	Entry             uint64
 	ReserveMemorySize uint64
 }
 
+// ZirconKernel abstracts a Zircon kernel file.
```

Comment:
If I understand [image.h](https://cs.opensource.google/fuchsia/fuchsia/+/main:zircon/system/public/zircon/boot/image.h) correctly `ZirconKernel` abstracts kernel loaded into memory, while `ZBIKernel` specifies Zircons kernel entry point and entrypoint.

Reply:
In [boot-zbi.h](https://cs.opensource.google/fuchsia/fuchsia/+/main:zircon/kernel/phys/include/phys/boot-zbi.h;l=102) `ZBIKernel` addressed as kernel header and `ZirconKernel` as kernel image. I find the this naming situation quite confusing.

Reply:
I'll just copy its current comment then, `// The whole contiguous image loaded into memory by the boot loader.`